allennlp/training/trainer.py

```diff
@@ -107,14 +110,21 @@ def train(self) -> None:
                     parameter.register_hook(clip_function)
 
         logger.info("Beginning training.")
+        num_training_batches = self._iterator.get_num_batches(self._train_dataset)
+        if self._validation_dataset is not None:
+            num_validation_batches = self._iterator.get_num_batches(self._validation_dataset)
         for epoch in range(epoch_counter, self._num_epochs):
             train_loss = 0.0
             val_loss = 0.0
             validation_metric_per_epoch = []  # type: List[float]
             # Set the model to "train" mode.
             self._model.train()
             train_generator = self._iterator(self._train_dataset, num_epochs=1)
-            for batch in tqdm.tqdm(train_generator):
+
+            train_progress_bar = tqdm.tqdm(train_generator, total=num_training_batches)
```

Comment:
`train_generator_with_progress_bar`?  That seems too long.  `train_generator_with_bar`?  `train_generator_with_progress`?  `train_generator_tqdm`? `tqdm_train_generator`?  I don't know, I don't much like any of these...